### PR TITLE
Enhance CSM

### PR DESCRIPTION
**Pull request by vhirtham**

### CODE REVIEW

File: weldx/transformations.py [hunk 1]
```diff
@@ -1078,19 +1106,56 @@ def add_cs(
...
-        self._add_coordinate_system_node(coordinate_system_name)
-        self._add_edges(
-            coordinate_system_name, reference_system_name, local_coordinate_system
-        )
+        if self.has_coordinate_system(coordinate_system_name):
+            # todo:
+            #  discuss: update and add functionality should be separated
+            #  why?   : to prevent errors. Misspelling of the system name might cause
+            #           unwanted updates or unwanted additions. Separate function can
+            #           catch that by knowing about the users intention.
```
Comment (vhirtham) [inline weldx/transformations.py:1126]: we should consider that

**Comment by CagtayFabry** [comment]
maybe we should use this PR to change the type hints for CS names from `Hashable` to `str` (no need for error catching)

File: weldx/transformations.py (hunk 1 above)
Comment (CagtayFabry) [reply weldx/transformations.py:1300]: having separate functions would complicate more "interactive" uses (change LCS but run same CSM code)

I think we can keep it like this for now, maybe add a nicer plot function to validate the CSM instance by the user instead :)

File: weldx/asdf/schemas/weldx.bam.de/weldx/core/transformations/coordinate_system_hierarchy_subsystem-1.0.0.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  members:
+    type: array
+    items:
+      type: string
+
+  subsystems:
+    type: array
+    items:
+      type: string
```
Comment (CagtayFabry) [inline weldx/asdf/schemas/weldx.bam.de/weldx/core/transformations/coordinate_system_hierarchy_subsystem-1.0.0.yaml:31]: add a little description to each property to clarify its meaning

File: weldx/asdf/schemas/weldx.bam.de/weldx/core/transformations/coordinate_system_hierarchy-1.0.0.yaml [hunk 3]
```diff
@@ -11,10 +11,24 @@ description: |
...
 properties:
+  name:
+    type: string
+
   root_system_name:
     description: |
       Name of the root coordinate system
     type: string
+
+  subsystems:
```
Comment (CagtayFabry) [inline weldx/asdf/schemas/weldx.bam.de/weldx/core/transformations/coordinate_system_hierarchy-1.0.0.yaml:22]:
```suggestion
  subsystem_names:
```

File: weldx/asdf/schemas/weldx.bam.de/weldx/core/transformations/coordinate_system_hierarchy-1.0.0.yaml [hunk 4]
```diff
@@ -11,10 +11,24 @@ description: |
...
+    type: array
+    items:
+      type: string
+
+  subsystem_data:
```
Comment (CagtayFabry) [inline weldx/asdf/schemas/weldx.bam.de/weldx/core/transformations/coordinate_system_hierarchy-1.0.0.yaml:27]:
```suggestion
  subsystems:
```

Comment (CagtayFabry) [reply weldx/asdf/schemas/weldx.bam.de/weldx/core/transformations/coordinate_system_hierarchy-1.0.0.yaml:27]: could we not use a single list? (list of names + named subsystems seems redundant)
Maybe with `wx_property_tag` validator?

**Review by CagtayFabry — APPROVED** [review]
Great work 👍 
We can do minor changes later :)

File: weldx/asdf/schemas/weldx.bam.de/weldx/core/transformations/coordinate_system_hierarchy-1.0.0.yaml (hunk 3 above)
Comment (vhirtham) [reply weldx/asdf/schemas/weldx.bam.de/weldx/core/transformations/coordinate_system_hierarchy-1.0.0.yaml:22]: Done

File: weldx/asdf/schemas/weldx.bam.de/weldx/core/transformations/coordinate_system_hierarchy_subsystem-1.0.0.yaml (hunk 2 above)
Comment (vhirtham) [reply weldx/asdf/schemas/weldx.bam.de/weldx/core/transformations/coordinate_system_hierarchy_subsystem-1.0.0.yaml]: Done

File: weldx/asdf/schemas/weldx.bam.de/weldx/core/transformations/coordinate_system_hierarchy-1.0.0.yaml (hunk 4 above)
Comment (vhirtham) [reply weldx/asdf/schemas/weldx.bam.de/weldx/core/transformations/coordinate_system_hierarchy-1.0.0.yaml]: The list with the names contains only direct child subsystems. Nested ones are excluded. I added a corresponding description to clarify that.

**Pull request merged**